pkg/virt-controller/watch/volume-migration/volume-migration.go

```diff
@@ -141,6 +146,34 @@ func ValidateVolumes(vmi *virtv1.VirtualMachineInstance, vm *virtv1.VirtualMachi
 			valid = false
 			continue
 		}
+
+		// DataVolumes with a no-csi storage class
+		if v.VolumeSource.DataVolume != nil {
+			dv, err := storagetypes.GetDataVolumeFromCache(vm.Namespace, v.VolumeSource.DataVolume.Name, dvStore)
+			if err != nil {
+				return err
+			}
+			if dv == nil {
+				return fmt.Errorf("the datavolume %s doesn't exist", v.VolumeSource.DataVolume.Name)
+			}
+
+			// if the dv is in succeeded state then it is safe to use since it has already been populated.
+			if dv.Status.Phase == cdiv1.Succeeded {
+				continue
+			}
+			pvc, err := storagetypes.GetPersistentVolumeClaimFromCache(vm.Namespace, dv.Name, pvcStore)
+			if err != nil {
+				return err
+			}
+			if pvc == nil {
+				return fmt.Errorf("the pvc %s doesn't exist", v.VolumeSource.DataVolume.Name)
```

Comment:
similar to the comment above about DataVolumes. Not sure this is an error per-se.

What is different in this case is that the VM controller is not directly watching PVCs. But is is watching the DataVolumes that own PVCs which should be enough to get notified. But this case is a little trickier

Reply:
Same comment as above